### PR TITLE
Add 5.4 kernel and switch to using it

### PR DESCRIPTION
**Pull request by kraj**
This should fix https://github.com/mirzak/meta-coral/issues/51

Also add WIP patches for linux-imx 5.15 support.

### CODE REVIEW

File: recipes-kernel/linux/linux-coral_5.4.bb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    file://0012-arm64-dts-imx8mq-phanbell-Disable-busfreq-to-avoid-s.patch \
+    file://0013-MLK-15307-2-clk-imx8mq-set-the-parent-clocks-of-PCIE.patch \
+    file://0014-arm64-dts-imx8mq-Set-ext_osc-to-1-as-default.patch \
+    file://0015-PCI-imx-Use-the-external-clock-as-REF_CLK-when-neede.patch \
+    file://0016-PCI-imx-Provide-a-clock-to-the-device-for-i.MX8MQ.patch \
+    file://0017-arm64-dts-imx8mq-phanbell-Enable-PCIe.patch \
+    file://0018-spi-spi-imx-Revive-cs-gpios-property-for-DT.patch \
+    file://0019-arm64-dts-imx8mq-phanbell-Enable-ECSPI1.patch \
+    file://0020-arm64-dts-imx8mq-phanbell-Add-gpio-pinmux-for-40-pin.patch \
+    file://0021-arm64-dts-imx8mq-phanbell-Enable-I2C-2-I2C-3.patch \
```
Comment (mirzak) [inline recipes-kernel/linux/linux-coral_5.4.bb:39]: Where are these patches from?

**Review by mirzak — APPROVED** [review]

**Comment by mirzak** [comment]
Got some errors when trying to build this PR:

```
WARNING: linux-imx-headers-5.15-r0 do_fetch: Failed to fetch URL git://github.com/nxp-imx/linux-imx.git;protocol=https;branch=lf-5.15.y, attempting MIRRORS if available
ERROR: linux-imx-headers-5.15-r0 do_fetch: Fetcher failure: Fetch command export PSEUDO_DISABLED=1; export DBUS_SESSION_BUS_ADDRESS="unix:path=/run/user/1000/bus"; export PATH="/home/admin/src/coral/build/tmp/sysroots-uninative/x86_64-linux/usr/bin:/home/admin/src/coral/sources/poky/scripts:/home/admin/src/coral/build/tmp/work/cortexa53-crypto-mx8m-fslc-linux/linux-imx-headers/5.15-r0/recipe-sysroot-native/usr/bin/aarch64-fslc-linux:/home/admin/src/coral/build/tmp/work/cortexa53-crypto-mx8m-fslc-linux/linux-imx-headers/5.15-r0/recipe-sysroot/usr/bin/crossscripts:/home/admin/src/coral/build/tmp/work/cortexa53-crypto-mx8m-fslc-linux/linux-imx-headers/5.15-r0/recipe-sysroot-native/usr/sbin:/home/admin/src/coral/build/tmp/work/cortexa53-crypto-mx8m-fslc-linux/linux-imx-headers/5.15-r0/recipe-sysroot-native/usr/bin:/home/admin/src/coral/build/tmp/work/cortexa53-crypto-mx8m-fslc-linux/linux-imx-headers/5.15-r0/recipe-sysroot-native/sbin:/home/admin/src/coral/build/tmp/work/cortexa53-crypto-mx8m-fslc-linux/linux-imx-headers/5.15-r0/recipe-sysroot-native/bin:/home/admin/src/coral/sources/poky/bitbake/bin:/home/admin/src/coral/build/tmp/hosttools"; export HOME="/home/admin"; LANG=C git -c gc.autoDetach=false -c core.pager=cat clone --bare --mirror https://github.com/nxp-imx/linux-imx.git /home/admin/src/coral/downloads//git2/github.com.nxp-imx.linux-imx.git --progress failed with exit code 255, no output
ERROR: linux-imx-headers-5.15-r0 do_fetch: Bitbake Fetcher Error: FetchError('Unable to fetch URL from any source.', 'git://github.com/nxp-imx/linux-imx.git;protocol=https;branch=lf-5.15.y')
ERROR: Logfile of failure stored in: /home/admin/src/coral/build/tmp/work/cortexa53-crypto-mx8m-fslc-linux/linux-imx-headers/5.15-r0/temp/log.do_fetch.1384066
ERROR: Task (/home/admin/src/coral/sources/meta-freescale/recipes-kernel/linux/linux-imx-headers_5.15.bb:do_fetch) failed with exit code '1'
```

**Comment by kraj** [comment]
These errors could be due to net access is it happening all the time ? For me this repo fetches ok

File: recipes-kernel/linux/linux-coral_5.4.bb (hunk 1 above)
Comment (kraj) [reply recipes-kernel/linux/linux-coral_5.4.bb:39]: https://github.com/raitosyo/linux-coral/

**Comment by mirzak** [comment]
> These errors could be due to net access is it happening all the time ? For me this repo fetches ok

Could be something on my end, but the error is persistent and it is the only package that fails.

Merging this anyway.

**Pull request merged**